tests/test_flow/test_persistence_random.py

```diff
@@ -210,53 +240,86 @@ def _add_binding_to_flow(self, binding):
-        if memoized_values_by_entity_name is None:
-            memoized_values_by_entity_name = {}
-        if entity_name in memoized_values_by_entity_name:
-            return memoized_values_by_entity_name[entity_name]
+    def _compute_model_value(
+        self,
+        entity_name,
+        context,
+    ):
+        if entity_name in context.query_memoized_values_by_entity_name:
+            return context.query_memoized_values_by_entity_name[entity_name]
+        if entity_name in context.memoized_values_by_entity_name:
+            return context.memoized_values_by_entity_name[entity_name]
 
         entity = self._entities_by_name[entity_name]
         binding = entity.binding
 
         if not binding.has_persisted_values:
-            dep_values = [
-                self._compute_model_value(
-                    dep_entity_name,
-                    memoized_values_by_entity_name=memoized_values_by_entity_name,
+            running_in_subprocess = (
+                context.parallel_execution_enabled and binding.should_persist
+            )
+            if running_in_subprocess:
+                dep_context = context.evolve(
+                    memoized_values_by_entity_name={},
+                    query_memoized_values_by_entity_name={},
                 )
+            else:
+                dep_context = context
+            dep_values = [
+                self._compute_model_value(dep_entity_name, dep_context)
                 for dep_entity_name in binding.dep_entity_names
             ]
             values_by_entity_name = binding.compute_and_save_values(dep_values)
-            self._descriptors_computed_by_model.append(binding.out_descriptor)
+            context.computed_descriptors.append(binding.out_descriptor)
 
         else:
             if binding.persisted_values_stale_ancestor is not None:
-                assert self._versioning_mode == "assist"
+                assert context.versioning_mode == "assist"
                 stale_ancestor = binding.persisted_values_stale_ancestor
                 raise CodeVersioningError("Binding is stale", stale_ancestor)
             values_by_entity_name = binding.persisted_values_by_entity_name
 
-        memoized_values_by_entity_name.update(values_by_entity_name)
+        if binding.should_memoize:
+            context.memoized_values_by_entity_name.update(values_by_entity_name)
+        elif not binding.should_persist:
+            context.query_memoized_values_by_entity_name.update(values_by_entity_name)
+
         return values_by_entity_name[entity_name]
 
     def _clear_called_descriptors(self):
         self._descriptors_computed_by_flow[:] = []
         self._descriptors_computed_by_model[:] = []
 
 
+@attr.s
+class ModelExecutionContext:
+    """
+    Represents the state associated with a modeled ``flow.get()`` call.
+    """
+
+    parallel_execution_enabled = attr.ib()
+    versioning_mode = attr.ib()
+    memoized_values_by_entity_name = attr.ib()
```

Comment:
nit: I think it would be easier to read if you renamed these variables to `memoized_flow_values_...` and `memoized_query_values_...`.

Reply:
You're right, that's probably better.